torchrl/envs/transforms/transforms.py

```diff
@@ -1718,18 +1719,23 @@ def __init__(
 
     def reset(self, tensordict: TensorDictBase) -> TensorDictBase:
         """Resets _buffers."""
-        _reset = tensordict.get(
-            "_reset",
-            None,
-        )
-        if _reset is None:
-            _reset = torch.ones(
-                tensordict.batch_size,
-                dtype=torch.bool,
-                device=tensordict.device
-                if tensordict.device is not None
-                else torch.device("cpu"),
+        _reset = (
+            tensordict.get(
+                "_reset",
+                torch.ones(
+                    self.parent.done_spec.shape
+                    if self.parent
+                    else tensordict.batch_size,
+                    dtype=torch.bool,
+                    device=tensordict.device
+                    if tensordict.device is not None
+                    else torch.device("cpu"),
+                ),
             )
+            .view(*tensordict.batch_size, -1)
+            .any(-1)
```

Comment:
why not `any(range(tensordict.batch_dims, tensor.ndim))` or similar, instead of doing 2 ops?

Reply:
any does not support mult dim, but can use sum! Thanks